cli/register/register.go

```diff
@@ -386,6 +406,207 @@ func DoIt(org, pattern, nodeIdTok, userPw, email, inputFile string, nodeOrgFromF
 
 }
 
+// RegistrationFailure attempts to unregister the node if a critical error is encountered during registration.
+// This function will not return. It ends with a call to cliutils.Fatal
+func RegistrationFailure() {
+	msgPrinter := i18n.GetMessagePrinter()
+
+	msgPrinter.Printf("Critical error encountered in registration. Attempting to undo registration steps to leave node in the unregistered state.")
+	msgPrinter.Println("")
+
+	retries := 3
+	for i := 0; i < retries; i++ {
+		unregister.DeleteHorizonNode(false, false, 15)
+		err := unregister.CheckNodeConfigState(15)
+		if err == nil {
+			break
+		}
+		if i < retries-1 {
+			msgPrinter.Printf("Error unregistering node. Retrying.")
+			msgPrinter.Println()
+			time.Sleep(time.Second * 5)
+		} else {
+			cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("Failed to unregister node."))
```

Comment:
If we get to this point, 3 retries didn't work, I think we should tell user that we are goin to try a deep clean and then call unregister.DeleteHorizonNode(false, true, 15)

Reply:
Added